### PR TITLE
silence some annoying warnings

### PR DESCRIPTION
**Pull request by Megamouse**
Some warnings were driving me nuts.

### CODE REVIEW

**Comment by CookiePLMonster** [comment]
Idea for the future: There are also warnings about `std::iterator` being deprecated and IIRC they're fairly spammy too.

**Comment by MSuih** [comment]
The iterator warnings are coming from withing the standard library, we can't solve them ourselves. Some warnings have already been disabled from CMake, someone should port those changes over to visual studio project files as well.
https://github.com/RPCS3/rpcs3/blob/e54438d3a7f07fed2366c1de7eeb286443787fe3/rpcs3/cmake_modules/ConfigureCompiler.cmake#L10-L16

**Comment by Megamouse** [comment]
I don't really want to disable them totally, since they help cleaning up forgotten code.
(apart from 4281, which is annoying)

File: rpcs3/rpcs3qt/cheat_manager.cpp [hunk 1]
```diff
@@ -913,11 +927,12 @@ void cheat_manager_dialog::do_the_search()
...
 void cheat_manager_dialog::update_cheat_list()
 {
-	u32 num_rows = 0;
+	size_t num_rows = 0;
 	for (const auto& name : g_cheat.cheats)
 		num_rows += g_cheat.cheats[name.first].size();
 
-	tbl_cheats->setRowCount(num_rows);
+	ASSERT(num_rows <= INT32_MAX);
+	tbl_cheats->setRowCount(static_cast<int>(num_rows));
```
Comment (elad335) [inline rpcs3/rpcs3qt/cheat_manager.cpp]: this does both the caet and the assert.
```suggestion
	tbl_cheats->setRowCount(::narrow<int>(num_rows));
```

**Pull request merged**